src/main/resources/assets/js/app/settings/wizard/ProjectReadAccessWizardStepForm.ts

```diff
@@ -14,47 +14,112 @@ import {ProjectReadAccess, ProjectReadAccessType} from '../data/project/ProjectR
 
 export class ProjectReadAccessWizardStepForm
     extends SettingDataItemWizardStepForm<ProjectViewItem> {
 
-    private readAccessRadioGroup: RadioGroup;
+    private readAccessRadioGroup?: RadioGroup;
 
-    private readAccessRadioGroupFormItem: FormItem;
+    private readAccessRadioGroupFormItem?: FormItem;
 
-    private principalsCombobox: PrincipalComboBox;
+    private principalsCombobox?: PrincipalComboBox;
+
+    private localeCombobox: LocaleComboBox;
 
     constructor() {
         super();
     }
 
-    layout(item: ProjectViewItem) {
+    layout(item: ProjectViewItem): Q.Promise<void> {
         if (!item) {
-            return;
+            return Q(null);
         }
 
-        const readAccess: ProjectReadAccess = item.getData().getReadAccess();
-        this.readAccessRadioGroup.setValue(readAccess.getType(), true);
+        const layoutPromises: Q.Promise<any>[] = [];
+
+        layoutPromises.push(this.layoutLanguage(item.getLanguage()));
+
+        if (item.isDefaultProject()) {
+            return Q.all(layoutPromises).spread<void>(() => Q<void>(null));
+        }
+
+        layoutPromises.push(this.layoutReadAccess(item.getReadAccess(), item.getPermissions()));
 
-        this.updateFilteredPrincipalsByPermissions(item.getPermissions());
+        return Q.all(layoutPromises).spread<void>(() => Q<void>(null));
+    }
+
+    private layoutLanguage(language: string): Q.Promise<void> {
+        if (!language) {
+            return Q(null);
+        }
+
+        return this.getLocales().then((locales: Locale[]) => {
+            const localeToSelect: Locale = this.getLocaleByLanguage(language, locales);
 
-        if (readAccess.getType() === ProjectReadAccessType.CUSTOM) {
-            this.enablePrincipalCombobox();
+            if (localeToSelect) {
+                this.localeCombobox.select(localeToSelect, false, true);
+            }
+
+            return Q(null);
+        });
+    }
 
-            new GetPrincipalsByKeysRequest(readAccess.getPrincipals()).sendAndParse().then((principals: Principal[]) => {
-                principals.forEach((principal: Principal) => {
-                    this.principalsCombobox.select(principal);
-                });
-            }).catch(DefaultErrorHandler.handle);
+    private getLocales(): Q.Promise<Locale[]> {
+        const localeLoader: LocaleLoader = <LocaleLoader>this.localeCombobox.getLoader();
+        if (localeLoader.isLoaded()) {
+            return Q(this.localeCombobox.getDisplayValues());
         }
+
+        return this.localeCombobox.getLoader().load();
+    }
+
+    private getLocaleByLanguage(language: string, locales: Locale[]): Locale {
+        let selectedLocale: Locale = null;
+
+        locales.some((locale: Locale) => {
+            if (locale.getId() === language) {
+                selectedLocale = locale;
+                return true;
+            }
+
+            return false;
+        });
+
+        return selectedLocale;
```

Comment:
this can be replaced with

`return locales.find((locale: Locale) => locale.getId() === language);`